integration/src/test/java/org/kie/hacep/PodAsLeaderSnapshotTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.hacep;
+
+import java.util.ArrayList;
+import java.util.ConcurrentModificationException;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.hacep.core.Bootstrap;
+import org.kie.hacep.core.infra.election.State;
+import org.kie.hacep.message.ControlMessage;
+import org.kie.hacep.message.SnapshotMessage;
+import org.kie.remote.RemoteKieSession;
+import org.kie.remote.TopicsConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.*;
+import static org.kie.remote.util.SerializationUtil.deserialize;
+
+public class PodAsLeaderSnapshotTest {
+    private final String TEST_KAFKA_LOGGER_TOPIC = "logs";
+    private final String TEST_TOPIC = "test";
+    private KafkaUtilTest kafkaServerTest;
+    private Logger logger = LoggerFactory.getLogger(PodAsLeaderTest.class);
+    private EnvConfig config;
+    private TopicsConfig topicsConfig;
+
+    @Before
+    public void setUp() throws Exception {
```

Comment:
This method and `tearDown` is a copy/paste from `PodAsLeaderTest`. Please create and abstract to avoid code duplication (and update other test classes too)

Reply:
ok